.github/workflows/overall-tests.yml

```diff
@@ -0,0 +1,59 @@
+name: Overall Test Workflow
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  tests:
+    if: github.event.review.state == 'approved'
+    timeout-minutes: 40
```

Comment:
I would increase the timeout, probably 60 minutes

Reply:
At this point, it's not needed. Tests are taking 14min to 30 min  depending on OS and quantity of jobs running. We can increase it in the future if this becomes a problem, but I think we should focus on optmizing the test run to take less time, since 14 ~ 30 min is already too much.